### PR TITLE
virsh_qemu_monitor_command: ensure to start vm only if it is not alive

### PR DESCRIPTION
**Pull request by balamuruhans**
starting vm which is running tracebacks with VMStartError as previous test
didn't clean the vm and destroy vm on cleanup for proper test execution.

Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by balamuruhans** [comment]
**Issue observed**
```
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR| Traceback (most recent call last):
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-55.0-py2.7.egg/avocado_vt/test.py", line 397, in _runTest
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR|     run_func(self, params, env)
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR|   File "/home/bala/avocado-fvt-wrapper/data/avocado-vt/test-providers.d/downloads/powerkvm-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_qemu_monitor_command.py", line 32, in run
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR|     vm.start()
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-55.0-py2.7.egg/virttest/libvirt_vm.py", line 2337, in start
2017-11-20 13:13:30,436 stacktrace       L0047 ERROR|     raise virt_vm.VMStartError(self.name, result.stderr.strip())
2017-11-20 13:13:30,437 stacktrace       L0047 ERROR| VMStartError: VM 'virt-tests-vm1-lifecycle' failed to start: error: Domain is already active
```

**Comment by balamuruhans** [comment]
**After Fix:**
```
JOB ID     : bf9d2578d8234d305a2a3e4cc9135bb1f5349a02
JOB LOG    : /home/bala/avocado-fvt-wrapper/results/job-2017-11-20T15.09-bf9d257/job.log
 (01/40) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native:  PASS (43.10 s)
 (02/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.kvm_info.vm_running.valid_domname:  PASS (32.99 s)
 (03/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.kvm_info.vm_running.valid_domid:  PASS (34.22 s)
 (04/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.kvm_info.vm_running.valid_domuuid:  PASS (33.02 s)
 (05/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.kvm_info.vm_paused.valid_domname:  PASS (33.48 s)
 (06/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.kvm_info.vm_paused.valid_domid:  PASS (32.70 s)
 (07/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.kvm_info.vm_paused.valid_domuuid:  PASS (34.16 s)
 (08/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.cpus_info.vm_running.valid_domname:  PASS (38.99 s)
 (09/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.cpus_info.vm_running.valid_domid:  PASS (34.04 s)
 (10/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.cpus_info.vm_running.valid_domuuid:  PASS (32.73 s)
 (11/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.cpus_info.vm_paused.valid_domname:  PASS (34.36 s)
 (12/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.cpus_info.vm_paused.valid_domid:  PASS (34.12 s)
 (13/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.cpus_info.vm_paused.valid_domuuid:  PASS (32.68 s)
 (14/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.mem_info.vm_running.valid_domname:  PASS (33.23 s)
 (15/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.mem_info.vm_running.valid_domid:  PASS (34.29 s)
 (16/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.mem_info.vm_running.valid_domuuid:  PASS (38.95 s)
 (17/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.mem_info.vm_paused.valid_domname:  PASS (32.96 s)
 (18/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.mem_info.vm_paused.valid_domid:  PASS (32.74 s)
 (19/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.hmp_option.mem_info.vm_paused.valid_domuuid:  PASS (34.05 s)
 (20/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.qmp_option.vm_running.valid_domname:  PASS (32.69 s)
 (21/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.qmp_option.vm_running.valid_domid:  PASS (34.01 s)
 (22/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.qmp_option.vm_running.valid_domuuid:  PASS (32.99 s)
 (23/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.qmp_option.vm_paused.valid_domname:  PASS (32.87 s)
 (24/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.qmp_option.vm_paused.valid_domid:  PASS (34.52 s)
 (25/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.qmp_option.vm_paused.valid_domuuid:  PASS (38.03 s)
 (26/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.pretty_qmp.vm_running.valid_domname:  PASS (32.76 s)
 (27/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.pretty_qmp.vm_running.valid_domid:  PASS (38.03 s)
 (28/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.pretty_qmp.vm_running.valid_domuuid:  PASS (38.07 s)
 (29/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.pretty_qmp.vm_paused.valid_domname:  PASS (33.16 s)
 (30/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.pretty_qmp.vm_paused.valid_domid:  PASS (33.10 s)
 (31/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.normal_test.pretty_qmp.vm_paused.valid_domuuid:  PASS (33.23 s)
 (32/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.invalid_domname:  PASS (32.88 s)
 (33/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.invalid_domid:  PASS (33.05 s)
 (34/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.hex_domid:  PASS (33.54 s)
 (35/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.invalid_domuuid:  PASS (34.19 s)
 (36/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.none_domname:  PASS (33.27 s)
 (37/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.invalid_options:  PASS (33.83 s)
 (38/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.unknown_cmd:  PASS (33.26 s)
 (39/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.invalid_options.invalid_qmp_cmd:  PASS (33.23 s)
 (40/40) domstat.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.0.ppc64le.powerkvm-libvirt.virsh.qemu_monitor_command.error_test.vm_shutoff:  PASS (0.97 s)
RESULTS    : PASS 40 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1352.11 s
JOB HTML   : /home/bala/avocado-fvt-wrapper/results/job-2017-11-20T15.09-bf9d257/results.html
```

File: libvirt/tests/src/virsh_cmd/domain/virsh_qemu_monitor_command.py [hunk 1]
```diff
@@ -25,12 +23,13 @@ def run(test, params, env):
...
         if "--pretty" not in help_info:
-            raise error.TestNAError("--pretty option is not supported in"
-                                    " current version")
+            test.cancel("--pretty option is not supported in current version")
 
     try:
         # Prepare vm state for test
-        if vm_state != "shutoff":
+
+        # start vm if it is not alive
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_qemu_monitor_command.py]: start to Start

**Comment by balamuruhans** [comment]
@chunfuwen done

**Pull request merged**